File: Himalayan/Himalayan/Controllers/Base/BaseController.m
//
//  BaseController.m
//  FddOrder
//
//  Created by ysjjchh on 14/11/24.
//  Copyright (c) 2014年 fangdd. All rights reserved.
//

#import "BaseController.h"

@interface BaseController () <UITextFieldDelegate>

@property (nonatomic, strong) UISwipeGestureRecognizer *swipe;

@end

@implementation BaseController

- (instancetype)init
{
    self = [super init];
    if (self) {
//        self.hidesBottomBarWhenPushed = YES;
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    if ([self systemVersion] < 7) {
        self.view.height -= 44;
    } else {
        self.view.height -= 64;
        self.edgesForExtendedLayout = UIRectEdgeNone;
    }
    
//    self.view.backgroundColor = UIColorWithRGBA(235, 235, 241, 1);
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)dealloc {
    NSLog(@"%@ dealloc", [self class]);
}

- (void)onSwipe {
    if (!self.navigationItem.hidesBackButton) {
        if (!self.navigationItem.hidesBackButton) {
            if (self.navigationController.viewControllers.count > 1) {
                [self.navigationController popViewControllerAnimated:YES];
            }
        }
    }
}

- (void)removeSwipeGesture {
    if (_swipe) {
        [self.view removeGestureRecognizer:_swipe];
    }
}

- (UIButton *)showRightItemWithTitle:(NSString *)title {
    UIButton *button = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 80, 34)];
    [button setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    [button setTitle:title forState:UIControlStateNormal];
    button.contentHorizontalAlignment = UIControlContentHorizontalAlignmentRight;
    
    UIBarButtonItem *item = [[UIBarButtonItem alloc] initWithCustomView:button];
    self.navigationItem.rightBarButtonItem = item;
    
    return button;
}

- (UIButton *)showRightItemWithImage:(NSString *)imageName {
    UIButton *button = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 80, 34)];
    [button setImage:[UIImage imageNamed:imageName] forState:UIControlStateNormal];
    button.contentHorizontalAlignment = UIControlContentHorizontalAlignmentRight;
    
    UIBarButtonItem *item = [[UIBarButtonItem alloc] initWithCustomView:button];
    self.navigationItem.rightBarButtonItem = item;
    
    return button;
}

- (UIButton *)showLeftItemWithImage:(NSString *)imageName {
    UIButton *button = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 80, 34)];
    [button setImage:[UIImage imageNamed:imageName] forState:UIControlStateNormal];
    button.contentHorizontalAlignment = UIControlContentHorizontalAlignmentLeft;
    
    UIBarButtonItem *item = [[UIBarButtonItem alloc] initWithCustomView:button];
    self.navigationItem.leftBarButtonItem = item;
    
    return button;
}

- (UIButton *)setBackItem {
    UIButton *button = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 40, 34)];
    [button setImage:[UIImage imageNamed:@"ic_back"] forState:UIControlStateNormal];
    button.contentHorizontalAlignment = UIControlContentHorizontalAlignmentLeft;
    [button addTarget:self action:@selector(onBack) forControlEvents:UIControlEventTouchUpInside];
    
    UIBarButtonItem *item = [[UIBarButtonItem alloc] initWithCustomView:button];
    self.navigationItem.leftBarButtonItem = item;
    
    return button;
}

- (void)onBack {
    [self.navigationController popViewControllerAnimated:YES];
}







#pragma -mark UITextFieldDelegate


- (int)systemVersion {
 
    return [[[UIDevice currentDevice] systemVersion] intValue];
}


- (CGFloat)stataHeight
{
    return [[UIApplication sharedApplication] statusBarFrame].size.height;
}

- (CGFloat)navHeight
{
    return self.navigationController.navigationBar.frame.size.height;
    
}

@end

File: kaifangbaoIOS/KaiFangBao/Base/BaseController.swift
//
//  BaseController.swift
//  KaiFangBao
//
//  Created by ysjjchh on 16/3/23.
//  Copyright © 2016年 fangdd. All rights reserved.
//

import Foundation

class BaseController: UIViewController {
    
    lazy var hud: WSProgressHUD = {
        let _hud = WSProgressHUD(view: self.view)
        self.view.addSubview(_hud)
        return _hud
    }()
    
    let networkManager:AFHTTPSessionManager = {
        let network = AFHTTPSessionManager()
        network.requestSerializer = AFJSONRequestSerializer()
        network.requestSerializer.timeoutInterval = 20
        network.requestSerializer.setValue(ClientSwift().UserAgetStringSwift(), forHTTPHeaderField: "User-Agent")
        network.responseSerializer = AFJSONResponseSerializer()
        network.responseSerializer.acceptableContentTypes = Set<String>(["text/html", "application/json", "text/plain"])
        return network
    }()
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(self.resetHTTPHeader), name: kNotificationLoginSuccess, object: nil)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    deinit {
        self.cancelRequests()
        NSNotificationCenter.defaultCenter().removeObserver(self)
        print("\(self.classForCoder) dealloc")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = KFBUtility.commonBackgroundColor
        self.automaticallyAdjustsScrollViewInsets = false
        
        if kIsIpad {
            let gesture = UISwipeGestureRecognizer(target: self, action: #selector(BaseController.onSwipe))
            gesture.direction = .Right
            self.view.addGestureRecognizer(gesture)
        }
    }
    
    func resetHTTPHeader() -> Void{
        let p = ClientSwift()
        print(p)
        self.networkManager.requestSerializer.setValue(ClientSwift().UserAgetStringSwift(), forHTTPHeaderField: "User-Agent")
    }
    
    func onSwipe() {
        
        if !self.navigationItem.hidesBackButton &&
            self.navigationController?.viewControllers.count > 1 {
            
            self.navigationController?.popViewControllerAnimated(true)
        }
    }
    
    func cancelRequests() {
        self.networkManager.operationQueue.cancelAllOperations()
        self.networkManager.session.invalidateAndCancel()
    }
    
    func showRightItemWithTitle(title: String) -> UIButton {
        let button = UIButton(frame: CGRectMake(0, 0, 60, 34))
        button.setTitleColor(UIColor.whiteColor(), forState: .Normal)
        button.setTitle(title, forState: .Normal)
        button.contentHorizontalAlignment = .Right
        
        let item = UIBarButtonItem(customView: button)
        self.navigationItem.rightBarButtonItem = item
        
        return button
    }
    
    func showRightItemWithImage(image: UIImage) -> UIButton {
        let button = UIButton(frame: CGRectMake(0, 0, 60, 34))
        button.setImage(image, forState: .Normal)
        button.contentHorizontalAlignment = .Right
        
        let item = UIBarButtonItem(customView: button)
        self.navigationItem.rightBarButtonItem = item
        
        return button
    }
    
    func showLeftItemWithImage(image: UIImage) -> UIButton {
        let button = UIButton(frame: CGRectMake(0, 0, 60, 34))
        button.setImage(image, forState: .Normal)
        button.contentHorizontalAlignment = .Left
        
        let item = UIBarButtonItem(customView: button)
        self.navigationItem.leftBarButtonItem = item
        
        return button
    }
    
    func setBackItem() {
        let button = self.showLeftItemWithImage(UIImage(named: "ic_back")!)
        button.addTarget(self, action: #selector(self.onBack), forControlEvents: .TouchUpInside)
    }
    
    func onBack() {
        self.navigationController?.popViewControllerAnimated(true)
    }
}












